skglm/experimental/sqrt_lasso.py

```diff
@@ -0,0 +1,230 @@
+
+        coefs : array, shape (n_features, n_alphas)
+            Coefficients along the path.
+        """
+        if not hasattr(self, "solver_"):
+            self.solver_ = ProxNewton(
+                tol=self.tol, max_iter=self.max_iter, verbose=self.verbose)
+        # build path
+        if alphas is None:
+            alpha_max = norm(X.T @ y, ord=np.inf) / (np.sqrt(len(y)) * norm(y))
+            alphas = alpha_max * np.geomspace(1, eps, n_alphas)
+        else:
+            n_alphas = len(alphas)
+            alphas = np.sort(alphas)[::-1]
+
+        n_features = X.shape[1]
+        sqrt_quadratic = compiled_clone(SqrtQuadratic())
+        l1_penalty = compiled_clone(L1(1.))  # alpha is set along the path
+
+        coefs = np.zeros((n_alphas, n_features))
+
+        for i in range(n_alphas):
+            if self.verbose:
+                to_print = "##### Computing alpha %d/%d" % (i + 1, n_alphas)
+                print("#" * len(to_print))
+                print(to_print)
+                print("#" * len(to_print))
+
+            l1_penalty.alpha = alphas[i]
+            # no warm start for the first alpha
+            coef_init = coefs[i].copy() if i else np.zeros(n_features)
+
+            try:
+                coef, _, _ = self.solver_.solve(
+                    X, y, sqrt_quadratic, l1_penalty,
+                    w_init=coef_init, Xw_init=X @ coef_init)
+                coefs[i] = coef
+            except ValueError as val_exception:
+                # make sure to catch residual error
+                # it's implemented this way as Numba doesn't support custom Exception
+                if not str(val_exception) == "SmallResidualException":
+                    raise
+
+                # save coef despite not converging
+                # coef_init holds a ref to coef
+                coef = coef_init
+                res_norm = norm(y - X @ coef)
+                warnings.warn(
+                    f"Small residuals prevented the solver from converging "
+                    f"at alpha={alphas[i]:.2e} (residuals' norm: {res_norm:.4e}). "
+                    "Consider fitting with higher alpha.",
+                    ConvergenceWarning
+                )
+                coefs[i] = coef
+                break
+
+        return alphas, coefs
+
+
+def _chambolle_pock_sqrt(X, y, alpha, max_iter=1000, obj_freq=10):
+    """Apply Chambolle-Pock algorithm to solve square-root Lasso.
+
+    The objective function is:
+        min_w ||Xw - y||_2/sqrt(n_samples) + alpha * ||w||_1.
+    """
+    n_samples, n_features = X.shape
+    # dual variable is z, primal is w
+    z_old = np.zeros(n_samples)
+    z = z_old.copy()
+    w = np.zeros(n_features)
+
+    objs = []
+
+    L = norm(X, ord=2)
+    # take primal and dual stepsizes equal
+    tau = 0.99 / L
+    sigma = 0.99 / L
+
+    for t in range(max_iter):
+        w = ST_vec(w - tau * X.T @ (2 * z - z_old), alpha * np.sqrt(n_samples) * tau)
```

Comment:
This feels weird to see a `np.sqrt(n_samples)` term.
It feels very sqrt-Lasso-dependent, and feels that this will not generalize well to other datafits.

I do not know for this PR, but maybe we could try to isolate the generic Chambolle-Pock solver, and allow to change the default solver (Prox-Newton or Chambolle-Pock) in `SqrtLasso`.